pkg/backup/backup_compaction_test.go

```diff
@@ -340,6 +343,46 @@ crdb_internal.json_to_pb(
 			t, db, []string{"foo"}, "'nodelocal://1/backup/9'", start, end, opts,
 		)
 	})
+
+	t.Run("maybeStartCompactionJob", func(t *testing.T) {
+		db.Exec(t, "SET CLUSTER SETTING backup.compaction.threshold = 3")
+		db.Exec(t, "SET CLUSTER SETTING backup.compaction.window_size = 2")
+		db.Exec(t, "CREATE TABLE foo (a INT, b INT)")
+		defer func() {
+			db.Exec(t, "DROP TABLE foo")
+			db.Exec(t, "SET CLUSTER SETTING backup.compaction.threshold = 0")
+		}()
+		db.Exec(t, "INSERT INTO foo VALUES (1, 1)")
+		db.Exec(t, "BACKUP INTO 'nodelocal://1/backup/10'")
+		db.Exec(t, "INSERT INTO foo VALUES (2, 2)")
+		db.Exec(t, fmt.Sprintf(incBackupCmd, 10))
+		db.Exec(t, "INSERT INTO foo VALUES (3, 3)")
+		var incJobID jobspb.JobID
+		db.QueryRow(t, "BACKUP INTO LATEST IN 'nodelocal://1/backup/10' WITH detached").Scan(&incJobID)
+		waitForSuccessfulJob(t, tc, incJobID)
+		var jobPayload jobspb.Payload
+		var payloadBytes []byte
+		db.QueryRow(
+			t,
+			fmt.Sprintf(
+				`SELECT payload FROM crdb_internal.system_jobs WHERE id = %d`,
+				incJobID,
+			),
+		).Scan(&payloadBytes)
+		require.NoError(t, protoutil.Unmarshal(payloadBytes, &jobPayload))
+		backupDetails, ok := jobPayload.UnwrapDetails().(jobspb.BackupDetails)
+		require.True(t, ok, "expected job details to be of type jobspb.BackupDetails")
+		execCfg, _ := tc.Server(0).ExecutorConfig().(sql.ExecutorConfig)
+		jobID, err := maybeStartCompactionJob(
+			ctx,
+			&execCfg,
+			username.RootUserName(),
+			backupDetails,
+			fmt.Sprintf(incBackupCmd, 10),
+		)
+		require.NoError(t, err)
+		waitForSuccessfulJob(t, tc, jobID)
+	})
```

Comment:
Would like to add a nightly test that tests the entire pipeline of scheduled backups triggering compaction. For reasons outlined [here](https://cockroachlabs.slack.com/archives/C03JCUUSCD6/p1741899463662109), this is a bit of a hassle.

Reply:
Did I say a hassle? I lied, @msbutler had a much simpler solution